addOns/custompayloads/src/main/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsMultipleOptionsTablePanel.java

```diff
@@ -103,6 +122,69 @@ public void actionPerformed(ActionEvent evt) {
         addButton(resetButton);
     }
 
+    private void addPayloadFileButton() {
+        fileButton = new JButton(ADD_MULTIPLE_PAYLOADS_BUTTON);
+        fileButton.addActionListener(
+                e -> {
+                    CustomPayload multiplePayloads = new CustomPayload(-1, true, "", "");
+                    CustomMultiplePayloadDialog dialog =
+                            new CustomMultiplePayloadDialog(
+                                    View.getSingleton().getOptionsDialog(null), multiplePayloads);
+                    dialog.pack();
+                    dialog.setVisible(true);
+                    File file = null;
+                    boolean preventDuplicates = false;
+                    if (dialog.isSaved()) {
+                        file = dialog.getFile();
+                        preventDuplicates = dialog.isPreventDuplicates();
+                    }
+                    if (file == null) {
+                        return;
+                    }
+                    try (BufferedReader txtReader = Files.newBufferedReader(file.toPath())) {
+                        String line;
+                        ArrayList<CustomPayload> payloads = new ArrayList<>();
+                        Set<String> existingPayloads = new HashSet<>();
+                        if (preventDuplicates) {
+                            tableModel.getPayloadsOfACategory(
+                                    existingPayloads, multiplePayloads.getCategory());
+                        }
+                        while ((line = txtReader.readLine()) != null) {
+                            CustomPayload newPayload =
+                                    new CustomPayload(multiplePayloads.getCategory(), "");
+                            newPayload.setPayload(line);
+                            if (preventDuplicates) {
+                                boolean alreadyExisting = false;
+                                for (String existingPayload : existingPayloads) {
+                                    if (newPayload.getPayload().equals(existingPayload)) {
+                                        alreadyExisting = true;
+                                        break;
+                                    }
+                                }
+                                if (!alreadyExisting) {
+                                    payloads.add(newPayload);
+                                }
```

Comment:
A `contains` is more appropriate than iterate all of them but since we don't want duplicates we should also check that we are not importing duplicated ones, so we could just try add, e.g.:
```
if (existingPayloads.add(newPayload.getPayload())) {
    payloads.add(newPayload);
}